app/services/idv/steps/doc_auth_base_step.rb

```diff
@@ -40,13 +40,22 @@ def idv_failure(result)
       end
 
       def save_proofing_components
+        return unless current_user
```

Comment:
Do we know when this would be the case? I was wondering about hybrid flow, where in other places we use the `EffectiveUser` mixin to manage the user associated with the anonymous session. But are we ever actually calling `save_proofing_components` in that hybrid session (i.e. do we need to bother with that mixin?)?

Reply:
I am hoping it was just an artifact of how sloppily the individual feature tests for steps were done, it doesn't make sense why we need this check

Reply:
Curious if it's still needed with the recent round of revisions. Thinking the `ProofingComponent.create_or_find_by` would be more tolerant to the absence of `current_user`.

Reply:
There's a null constraint on the column so this blows up:

```ruby
[1] pry(main)> ProofingComponent.create_or_find_by(user: nil)
  TRANSACTION (0.1ms)  BEGIN
  ProofingComponent Create (2.4ms)  INSERT INTO "proofing_components" ("created_at", "updated_at") VALUES ($1, $2) RETURNING "id"  [["created_at", "2021-12-21 22:05:23.908635"], ["updated_at", "2021-12-21 22:05:23.908635"]]
  TRANSACTION (0.2ms)  ROLLBACK
ActiveRecord::NotNullViolation: PG::NotNullViolation: ERROR:  null value in column "user_id" of relation "proofing_components" violates not-null constraint
DETAIL:  Failing row contains (9, null, null, null, null, null, null, null, 2021-12-21 22:05:23.908635, 2021-12-21 22:05:23.908635, null).
```